esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -5919,6 +5919,10 @@ void CWsDfuEx::getFilePartsInfo(IEspContext &context, IFileDescriptor &fileDesc,
         Owned<IEspDFUFilePart> filePart = createDFUFilePart();
         filePart->setPartIndex(part.queryPartIndex() + 1);
         filePart->setCopies(fileCopies);
+
+        if (version > 1.55) //not sure if this is needed
+            filePart->setTopLevelKey(isPartTLK(&part));
```

Comment:
If <= 1.55 doesn't want to know about key part type, then not sure what it's going to do with an extra part that it will think is a regular part and will get spurious invalid rows back from..
I think needs to suppress the TLK's parts until the client knows what to do with them, which can initially be to ignore them.

Reply:
yep, agreed.